### ISSUE TITLE
Imported file size optimization

### ISSUE THREAD

**Issue by SickSAMA** [searchkit/searchkit#305]
Thank you for creating such a powerful tool and it really saves a lot of time in our project. 

Our project uses webpack 2 and ES6, the final minified js file is too big (751kb) and I inspected the code and found that Searchkit imports everything into the final js file. The entry point of searchkit is `lib/src/index.js` which exports all components. Even though I only import the components I use, the unused components are still imported because of that. Webpack 2's treeshake feature only support es6 module syntax so that does not help.

 I tried to import components directly from their source file like 

```
import { RefinementListFilter } from 'searchkit/lib/src/components/search/filters/facet-filter/RefinementListFilter';
```

But I came across the error `MenuFilter.js:5 Uncaught TypeError: Cannot read property 'prototype' of undefined`. The wield thing is that I didn't import MenuFilter at all. 

Searchkit uses a lot of syntax like

```
import {
   CheckboxItemList, Panel
} from "../../../ui"
```

This is convenient for development but build too many links between unrelated files. The above code which is in `FacetFilter.tsx` connect itself to all ui components. Even though I directly import `FacetFilter.js`, all ui components are also imported.

**Comment by GregoryPotdevin** [comment]
I'll have to try out Webpack2, but shouldn't 
```
import {
   CheckboxItemList, Panel
} from "../../../ui"
```
work with Webpack's tree-shaking ?

**Comment by SickSAMA** [comment]
Webpack's tree-shaking only supports es6 import and export. Modules imported from ```node_modules``` are exported using commonjs.

**Comment by GregoryPotdevin** [comment]
Ok. I compared it to other projects I released on npm (like searchkit-daterangefilter), which are built and published using nwb. Those actually have 3 output directories :
- lib => multiple JS files, commonjs ?)
- es6 => multiple JS files, using import/export
- umd => single JS file

That's probably what we need for searchkit. The package.json files I use in those projects also contains 
```
{
    "main": "lib/index.js",
    "jsnext:main": "es6/index.js"
}
```

I guess that's how babel knows which one to use ?

@ssetem I have some times this week if you want us to check that out
